pkg/messaging/rabbitmq/pubsub.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package rabbitmq
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+// SubjectAllChannels represents subject to subscribe for all the channels.
+const (
+	ChansPrefix        = "channels"
+	SubjectAllChannels = "channels.>"
+	RoutingKey         = "application specific routing key for fancy topologies"
+	Exchange           = "mainflux"
+	ExchangeKind       = "fanout"
+	QueueDurability    = true
+	QueueDelete        = false
+	QueueExclusivity   = false
+	QueueWait          = false
+	ConsumerTag        = "mainflux-consumer"
+	Mandatory          = false
+	Immediate          = false
+)
+
+var (
+	errAlreadySubscribed = errors.New("already subscribed to topic")
+	errNotSubscribed     = errors.New("not subscribed")
+	errEmptyTopic        = errors.New("empty topic")
+)
+
+var _ messaging.PubSub = (*pubsub)(nil)
+
+// PubSub wraps messaging Publisher exposing
+// Close() method for RabbitMQ connection.
+type PubSub interface {
+	messaging.PubSub
+	Close()
+}
+
+type pubsub struct {
+	conn          *amqp.Connection
+	logger        log.Logger
+	queue         amqp.Queue
+	channel       *amqp.Channel
+	subscriptions map[string]bool
+	done          chan bool
```

Comment:
Is this used?

Reply:
No

Reply:
What for, what's the purpose of the `done` channel?